components/src/components/top-button/top-button.tsx

```diff
@@ -0,0 +1,31 @@
+import { Component, Listen, Prop, h } from '@stencil/core';
+
+// Scroll to top button.
+@Component({
+  tag: 'pulumi-top-button',
+  styleUrl: 'top-button.scss',
+  shadow: false
+})
+
+export class TopButton {
+
+  @Prop({ mutable: true })
+  scrollPostion: number;
+
+  @Listen('scroll', { target: 'window' })
+  handleScroll() {
+    this.scrollPostion = window.scrollY;
+  }
+
+  componentWillRender() {
+    this.scrollPostion = window.scrollY;
+  }
+
+  render() {
+    return (
+      <div>
+          <a class="top-button fas fa-chevron-up" title="Scroll to top" style={{display: this.scrollPostion < 2500 ? "none" : ""}}></a>
```

Comment:
```
style={{display: this.[...]
```

TIL -- I had no idea Stencil/JSX had types defined at the style-tag level. 🎓

Reply:
My approach, for better or worse, has just been pretending stencil is react, until stencil screams at me. lol.